server/account/account.controller.js

```diff
@@ -17,84 +17,91 @@ const {
   406 => Not Acceptable
 */
 module.exports.register = async (req, res, next) => {
-	try {
-		const data = req.body;
-		const { error } = signupDataValidation(data);
-		if (error)
-			return res.status(400).json(error.details.map(err => err.message));
-		const { role } = data;
-		if (role !== "admin" && role !== "researcher" && role !== "audience") {
-			return res.status(400).json({
-				message: 'Choose designation between "admin", "researcher" & "audience"'
-			});
-		}
-		let user = await AccountModel.findOne({ email: data.email });
-		if (user) return res.status(400).json({ message: "Email already Exists" });
-		const hashedPassword = await bcrypt.hash(data.password, 10);
-		const newUser = new AccountModel({
-			name: data.name,
-			email: data.email,
-			phoneNumber: data.phoneNumber,
-			role: data.role,
-			password: hashedPassword
-		});
-		await newUser.save();
-		return res
-			.status(200)
-			.json({ message: "Registration Successful", data: newUser });
-	} catch (error) {
-		return res.status(500).json({ message: "Internal Server Error" });
-	}
+  try {
+    const data = req.body;
+    const { error } = signupDataValidation(data);
+    if (error)
+      return res.status(400).json(error.details.map(err => err.message));
+    const { role } = data;
+    if (role !== "admin" && role !== "researcher" && role !== "audience") {
+      return res.status(400).json({
+        message: 'Choose designation between "admin", "researcher" & "audience"'
+      });
+    }
+    let user = await AccountModel.findOne({ email: data.email });
+    if (user) return res.status(400).json({ message: "Email already Exists" });
+    console.log(data, user);
+    user = await AccountModel.findOne({ phoneNumber: data.phoneNumber });
+    if (user)
+      return res.status(400).json({ message: "Phone Number already Exists" });
+    console.log(data, user);
+
+    const hashedPassword = await bcrypt.hash(data.password, 10);
+    const newUser = new AccountModel({
+      name: data.name,
+      email: data.email,
+      phoneNumber: data.phoneNumber,
+      role: data.role,
+      password: hashedPassword
+    });
+    await newUser.save();
+    newUser.password = undefined;
```

Comment:
Very Good thing @LetsTrie

Reply:
It will undefined when it is returning the object.